File: src/main/java/edu/marmara/dto/CourseGetDTO.java
package edu.marmara.dto;

import java.util.List;

public class CourseGetDTO {
    private String courseCode;
    private String courseTitle;
    private Integer givenSemester;
    private Integer courseCredit;
    private List<String> prerequisites;
    private List<String> weeklyDate;
    private Integer maxSeats;
    private Integer takenSeats;
    private Integer minCreditReq;

    public CourseGetDTO() {
    }

    public CourseGetDTO(String courseCode, String courseTitle, Integer givenSemester, Integer courseCredit, List<String> prerequisites, List<String> weeklyDate, Integer maxSeats, Integer takenSeats, Integer minCreditReq) {
        this.courseCode = courseCode;
        this.courseTitle = courseTitle;
        this.givenSemester = givenSemester;
        this.courseCredit = courseCredit;
        this.prerequisites = prerequisites;
        this.weeklyDate = weeklyDate;
        this.maxSeats = maxSeats;
        this.takenSeats = takenSeats;
        this.minCreditReq = minCreditReq;
    }

    @Override
    public String toString() {
        return "CourseGetDTO{" +
                "courseCode='" + courseCode + '\'' +
                ", courseTitle='" + courseTitle + '\'' +
                ", givenSemester=" + givenSemester +
                ", courseCredit=" + courseCredit +
                ", prerequisites=" + prerequisites +
                ", weeklyDate=" + weeklyDate +
                '}';
    }

    public String getCourseCode() {
        return courseCode;
    }

    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    public String getCourseTitle() {
        return courseTitle;
    }

    public void setCourseTitle(String courseTitle) {
        this.courseTitle = courseTitle;
    }

    public Integer getGivenSemester() {
        return givenSemester;
    }

    public void setGivenSemester(Integer givenSemester) {
        this.givenSemester = givenSemester;
    }

    public Integer getCourseCredit() {
        return courseCredit;
    }

    public void setCourseCredit(Integer courseCredit) {
        this.courseCredit = courseCredit;
    }

    public List<String> getPrerequisites() {
        return prerequisites;
    }

    public void setPrerequisites(List<String> prerequisites) {
        this.prerequisites = prerequisites;
    }

    public List<String> getWeeklyDate() {
        return weeklyDate;
    }

    public void setWeeklyDate(List<String> weeklyDate) {
        this.weeklyDate = weeklyDate;
    }

    public Integer getMaxSeats() {
        return maxSeats;
    }

    public void setMaxSeats(Integer maxSeats) {
        this.maxSeats = maxSeats;
    }

    public Integer getTakenSeats() {
        return takenSeats;
    }

    public void setTakenSeats(Integer takenSeats) {
        this.takenSeats = takenSeats;
    }

    public Integer getMinCreditReq() {
        return minCreditReq;
    }

    public void setMinCreditReq(Integer minCreditReq) {
        this.minCreditReq = minCreditReq;
    }
}

File: src/main/marmara/dto/CourseGetDTo.py
from typing import List


class CourseGetDTO:
    def __init__(self, course_code: str, course_title: str, given_semester: int, course_credit: int,
                 prerequisites: List[str], weekly_date: List[str], max_seats: int, taken_seats: int,
                 min_credit_req: int):
        self.course_code = course_code
        self.course_title = course_title
        self.given_semester = given_semester
        self.course_credit = course_credit
        self.prerequisites = prerequisites
        self.weekly_date = weekly_date
        self.max_seats = max_seats
        self.taken_seats = taken_seats
        self.min_credit_req = min_credit_req

    def __repr__(self):
        return (f"CourseGetDTO(course_code='{self.course_code}', course_title='{self.course_title}', "
                f"given_semester={self.given_semester}, course_credit={self.course_credit}, "
                f"prerequisites={self.prerequisites}, weekly_date={self.weekly_date})")

    @property
    def course_code(self) -> str:
        return self._course_code

    @course_code.setter
    def course_code(self, course_code: str):
        self._course_code = course_code

    @property
    def course_title(self) -> str:
        return self._course_title

    @course_title.setter
    def course_title(self, course_title: str):
        self._course_title = course_title

    @property
    def given_semester(self) -> int:
        return self._given_semester

    @given_semester.setter
    def given_semester(self, given_semester: int):
        self._given_semester = given_semester

    @property
    def course_credit(self) -> int:
        return self._course_credit

    @course_credit.setter
    def course_credit(self, course_credit: int):
        self._course_credit = course_credit

    @property
    def prerequisites(self) -> List[str]:
        return self._prerequisites

    @prerequisites.setter
    def prerequisites(self, prerequisites: List[str]):
        self._prerequisites = prerequisites

    @property
    def weekly_date(self) -> List[str]:
        return self._weekly_date

    @weekly_date.setter
    def weekly_date(self, weekly_date: List[str]):
        self._weekly_date = weekly_date

    @property
    def max_seats(self) -> int:
        return self._max_seats

    @max_seats.setter
    def max_seats(self, max_seats: int):
        self._max_seats = max
